### PR TITLE
Dashboard Layout Implementation

### PR DESCRIPTION
**Pull request by dev-rathore**
_This PR contains changes related to the implementation of the dashboard layout and initial folder structure._

## Video
[crowdy-chat.webm](https://github.com/user-attachments/assets/0ba9c09f-c34b-4291-951c-d186337f9c31)

## Summary by CodeRabbit

- **New Features**
	- Introduced new layouts and components for authentication, campaigns, leads, sources, and settings.
	- Implemented a sidebar menu for improved navigation.
	- Added a custom logo and tab bar icon components.

- **Enhancements**
	- Enhanced the AppLayout for consistent theming and layout.
	- Updated the RootLayout to integrate authentication context.
	- Improved responsive design with new breakpoints and storage management capabilities.

- **Bug Fixes**
	- Resolved issues related to navigation and display of header elements.

- **Documentation**
	- Provided clear exports for context and storage management functionalities.

- **Chores**
	- Updated dependencies to improve navigation and secure storage capabilities.

### CODE REVIEW

**Comment by rahul-sinha1908** [comment]
@coderabbitai review

File: app/(auth)/login.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import AppLayout from "@/layouts/app-layout";
```
Comment (rahul-sinha1908) [inline app/(auth)/login.tsx:1]: @Manan-Sharma-5 Your code needs to be placed in this folder

**Comment by dev-rathore** [comment]
@coderabbitai review

**Review by Manan-Sharma-5 — APPROVED** [review]
Looks good to me

**Review by rahul-sinha1908 — APPROVED** [review]

**Pull request merged**